### PR TITLE
Don't export Props types from Functional Components

### PR DESCRIPTION
**Pull request by Vadorequest**
See https://alainperkaz-51714.medium.com/i-really-like-the-extension-of-props-there-i-may-update-some-of-my-storybook-stories-with-that-be524e76b082

### CODE REVIEW

File: src/common/components/animations/AnimatedTextBubble.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export type Props = {
   fill?: string;
 };
 
-const AnimatedTextBubble = (props: Props) => {
```
Comment (aperkaz) [inline src/common/components/animations/AnimatedTextBubble.tsx:16]: `React.FunctionComponent<Props>` adds `children` as an optional property. If the component doesnt support `children` prop, it adds an unnecessary property to the component.

I personally prefer setting `children` as an explicit property when declaring the property types. Doing so, the components that support children are always explicit about it.

```javascript
// example of prop types for component with children
type Props = {
  fill?: string;
  children: React.ReactNode;
}
```

**Review by aperkaz — COMMENTED** [review]
@Vadorequest PR looks good, its great to see that others are applying what I mentioned in the article 🙂 

Just a small note, `React.FunctionComponent<Props>` adds `children` as an optional property. If the component doesnt support children prop (or shouldent), it adds an unnecessary property to the component.

I personally prefer setting `children` as an explicit property when declaring the property types. Doing so, the components that support children are always explicit about it.

```javascript
// example of prop types for component with children
type Props = {
  fill?: string;
  children: React.ReactNode;
}
```

**Comment by Vadorequest** [comment]
@aperkaz Yeah, I agree, I prefer being explicit too about `children`. I'm not sure what I'm supposed to change here, though.

**Comment by aperkaz** [comment]
@Vadorequest instead of declaring component types with `React.FC<Props>` which adds `children` as an optional props, consider the following:

```javascript
// src/modules/core/errorHandling/DefaultErrorLayout.tsx

type Props = {
  error: Error;
  context?: GenericObject;
}

// OLD line, add children to the props, when its not used
// const DefaultErrorLayout: React.FunctionComponent<Props> = (props): JSX.Element => {

// NEW line, without optional children prop
const DefaultErrorLayout = (props: Props): JSX.Element => {
  const {
    error,
    context,
  } = props;
  const errorEventId = Sentry.captureException(error);
  return (
    <div
      css={css`
        text-align: center;
        margin-top: 30px;
        margin-bottom: 30px;
        .title {
          margin-top: 30px;
          margin-bottom: 30px;
        }
      `}
    >
      <div className={'title'}>
        <h1>Service currently unavailable</h1>
        <pre>Error 500.</pre>
      </div>
      <div>
        <p>
          Try to refresh the page. Please contact our support below if the issue persists.
        </p>
        <Button
          color={'primary'}
          onClick={(): void =>
            // @ts-ignore XXX showReportDialog is not recognised by TS (due to the webpack trick that replaces @sentry/node), but it works just fine
            Sentry.showReportDialog({ eventId: errorEventId })
          }
        >
          Contact support
        </Button>
      </div>
      {
        process.env.NEXT_PUBLIC_APP_STAGE !== 'production' && (
          <ErrorDebug
            error={error}
            context={context}
          />
        )
      }
    </div>
  );
};
export default DefaultErrorLayout;

```

**Comment by Vadorequest** [comment]
I see, I won't do it because it's a wide change that will cause many conflict for very little benefits.

What I'd rather do instead is defined my own `FunctionalComponentWithoutChildren` type, (or similar) and use that instead, but same, lots of changes for little benefit.

I'll merge it as it then, and keep it in mind for the next time. Thanks for your advises!

**Pull request merged**